src/effects/effectbuttonparameterslot.cpp

```diff
@@ -17,14 +17,16 @@ EffectButtonParameterSlot::EffectButtonParameterSlot(EffectsManager* pEffectsMan
     QString itemPrefix = formatItemPrefix(iParameterSlotNumber);
     m_pControlLoaded = new ControlObject(
             ConfigKey(m_group, itemPrefix + QString("_loaded")));
+
     m_pControlValue = new ControlPushButton(
             ConfigKey(m_group, itemPrefix));
-    m_pControlValue->setButtonMode(ControlPushButton::POWERWINDOW);
-    m_pControlType = new ControlObject(
-            ConfigKey(m_group, itemPrefix + QString("_type")));
-
     connect(m_pControlValue, SIGNAL(valueChanged(double)),
             this, SLOT(updateEngineState()));
+    ControlPushButton* pControlValue = static_cast<ControlPushButton*>(m_pControlValue);
```

Comment:
Huh, I didn't expect this to work because I did not realize `static_cast` implicitly calls the constructor. Please use `m_pControlValue = new ControlPushButton` instead for clarity.